lib/source/file.js

```diff
@@ -0,0 +1,169 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const fs = require('fs');
+const path = require('path');
+
+class FileParser {
+  constructor() {
+    this.properties = Object.create(null);
+  }
+
+  /**
+   * Parse the file contents
+   * @param {Buffer} data
+   */
+  update(data) {
+    let properties = {};
+
+    try {
+      properties = JSON.parse(data.toString());
+    } catch (e) {
+      // We should have an error condition but tbd
+    }
+    this.properties = properties.properties;
+  }
+}
+
+class File extends EventEmitter {
+  constructor(options) {
+    super();
+    this.configure(options);
+
+    this.type = 'file';
+    this.name = this.path;
+    this.properties = Object.create(null);
+    this._okay = false;
+    this._updated = null;
+    this._parser = new FileParser();
+    this.service = null;
+  }
+
+  /**
+   * Configures the file plugin
+   *
+   * The following options are valid:
+   * - path (String) The path to the file
+   *
+   * @param {Object} options Options that can be set for the plugin
+   */
+  configure(options) {
+    const params = options || Object.create(null);
+
+    if (!params.path) {
+      throw new Error('No path supplied');
+    }
+
+    try {
+      fs.accessSync(params.path, fs.R_OK);
+      this.path = params.path;
+    } catch (err) {
+      this._error(err);
+      throw err;
+    }
+  }
+
+  /**
+   * Clear the underlying properties object
+   */
+  clear() {
+    this.properties = Object.create(null);
+  }
+
+  initialize() {
+    if (this.service) {
+      return;
+    }
+
+    this._update(this.path);
+
+    this.service = fs.watch(this.path, {persistent: true, recursive: false}, (event, filename) => {
+      this._update(path.resolve(path.dirname(this.path), filename));
```

Comment:
Awesome, thanks for finding that. I think we should consider a moved file to be an error case we're not going to handle and simply make sure that `this.path` exists whenever we do an update.